### DISCUSSION TITLE
Method of Testing New Topic Contributions

### DISCUSSION THREAD

**Discussion by SkyScraperC** [Monika-After-Story/MonikaModDev#8681] [QA - code and content creation (technical)] [answered]
Hey all!

I got into DDLC back around in November, and needless to say I got hooked pretty quickly.
Monika isn't my _favourite_ Doki... but "Monika After Story" has definitely taken off way more than "Forever and Ever" has... She is my second favourite, so I'm happy loving Monika until I can find a way to spend more time with Sayori.

New to coding with Python, but I have some experience with coding in Java and C++ from University, so I think I've got the gist of what I need to do to contribute what I have in mind, but I digress...

_Anyway_, I'm hoping to contribute a new topic into "scripts-topics.rpy" and I've read all of the guidelines, I'll make a pull request when I'm ready, but I've got a fair start on what I want to add, and I'm struggling with testing what I've got so far (yes of course, because they all come up at random overtime, I only started MAS earlier about 5 days ago)

I've tried deleting all of the non-essential topics from a copy of "scripts-topics.rpy" and running it through the Ren'Py launcher and everything worked.... but even with the barebones in that file (and when I say barebones, I mean I removed everyone after line 845 in the current version of the code, and replaced it with my single topic), my topic wasn't coming up. Somehow, Monika kept bringing up different topics like video games and the fact that she's a vegetarian. I even tried assigning the action=EV_ACT_PUSH to the topic to increase its priority, but still no dice.

So, I'm just wondering what everyone else has done in terms of testing out their new codes of dialogue they wanted to add. Any suggestions? Is there any kind of dev tool that's been made which allows you to look at all of the different topics just to see how they would play out?

Kind of difficult to keep my motivation going on this contribution when I can't see what I have so far...

Also, I'd like to not mention the conversation topic itself, because I don't see anyone else having suggested it, and I'd rather not get opinions on it until its finished, because I _know_ the topic fits Monika's character, it's not memey, and it follows all of the guidelines posted.

Thanks for the help!

**Comment by Booplicate** [comment, marked as answer]
> my topic wasn't coming up. Somehow, Monika kept bringing up different topics like video games and the fact that she's a vegetarian. I even tried assigning the action=EV_ACT_PUSH to the topic to increase its priority, but still no dice.

We have *quite a few* topics, considering you just started, you most likely haven't seen most of them, so they are coming before yours. That's if you coded your topic properly.
Setting only `action` won't do anything. You need either `start_date` or `conditional` (or both) to trigger the action. Furthermore, changing that prop in your event definition *after* you already created the event object won't do anything. You need to modify the value in persistent (`action` one of the props that are being saved and loaded from persistent). There's multiple ways to do so. The most proper way would be using new persistent, that way you know how the event behaves when it first being seen. Although, you can get the event object and modify it.
Also, that action doesn't "increase" priority as is, it just shows the event when the condition is met.

> Also, I'd like to not mention the conversation topic itself, because I don't see anyone else having suggested it

1. If you're working on something, we can open an issue and assign you to it, if you're worried somebody would "steal" it from you.
2. Since your issue is not in the label, but in the event code, showing just that part would let people help you.

**Reply by SkyScraperC** [reply]
That's a great idea, thanks!

Everything I put after the prompt name in the event is:

            random=True,
            pool=False,
            action=EV_ACT_PUSH,
            #aff_range=(mas_aff.HAPPY, None)

I commented out the affection range for now knowing that I used a new Monika not wanting to mess up my own

And just for reference, this is what I meant when I said I replaced it with a "barebones" version of the file. After taking out most of the content, I was sure that mine would come up (and no, I'm not trying to load it as a .txt file, I just can't load a .rpy file to a post on github)

[script-topics.rpy.txt](https://github.com/Monika-After-Story/MonikaModDev/files/8011750/script-topics.rpy.txt)

**Reply by SkyScraperC** [reply]
But I'll give it another try after removing the action line and let you know what happens

**Reply by multimokia** [reply]
So, `action` won't actually work without a `conditional` or a `start_date`/`end_date`

You'll also need to edit the topic data manually now that it's in your persistent, as those fields are meant to not be overwritten on each load.

**Reply by Booplicate** [reply]
Yeah, `action` does nothing by itself. Your topic is random, so it may take a while for it to be selected.
You could add `conditional="True",` which would trigger the action and you'd get the topic, but you'd have to restore persistent or modify the event data.

**Reply by SkyScraperC** [reply]
So, now my event block looks something kind of like this:

persistent.event_database,
            eventlabel="monika________",
            category=["society"],
            prompt="____________________",
            random=True,
            #aff_range=(mas_aff.HAPPY, None)

and I started with yet another new Monika after you guys said there might be a conflict with the persistent file.

and she's already said:
"I hope you've enjoyed listening to everything I was thinking about today..."
"I'm sure I'll have something to talk about in a while"

but she never mentioned my topic....
Any reason why this block wouldn't have triggered with a new persistent file?

**Reply by SkyScraperC** [reply]
OKAY! It finally triggered... Might have had something to do with the fact that the new file persistent defaults to having repeat conversations? Not totally sure, but I'm glad I got to see it, and now I can just trigger it in repeat conversations if I want to see it again with this Monika... That took a while. Thanks for all of the help!!

**Comment by multimokia** [comment]
Additionally, [here's](https://github.com/Monika-After-Story/MonikaModDev/wiki/Dialogue-Coding#the-init-5-python-block) some documentation regarding to how the full event structure works in terms of its parameters and what they do

**Reply by SkyScraperC** [reply]
Thanks for the referral, but I already studied that page for quite a while, and it didn't really answer my question.
